service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomServiceImplTest.java

```diff
@@ -0,0 +1,299 @@
+package cz.muni.fi.pa165.mushrooms.service;
+
+import cz.muni.fi.pa165.mushrooms.dao.MushroomDao;
+import cz.muni.fi.pa165.mushrooms.dao.VisitDao;
+import cz.muni.fi.pa165.mushrooms.entity.Mushroom;
+import cz.muni.fi.pa165.mushrooms.enums.MushroomType;
+import mockit.Delegate;
+import mockit.Expectations;
+import mockit.Injectable;
+import mockit.Tested;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * Tests for service layer related to Mushroom entity
+ * Not much to test here actually, service is just relaying to DAO layer
+ *
+ * @author bencikpeter
+ */
+
+public class MushroomServiceImplTest {
+
+    @Injectable
+    private MushroomDao mushroomDao;
+
+    @Tested(fullyInitialized = true)
+    private MushroomServiceImpl service;
+
+    private class MockDatabase {
+        private Map<Long, Mushroom> database = new HashMap<>();
+        private long databaseCounter = 0;
+
+        private void validate(Mushroom mushroom){
+            if (mushroom == null) throw new IllegalArgumentException("null");
+            if (mushroom.getName() == null) throw new IllegalArgumentException("nameIsNull");
+            if (mushroom.getIntervalOfOccurrence() == null) throw new IllegalArgumentException("interval of occurence is null");
+            if (mushroom.getType() == null) throw new IllegalArgumentException("type is null");
+        }
+
+
+        public void create(Mushroom mushroom) {
+            validate(mushroom);
+            if (mushroom.getId() != null) throw new IllegalArgumentException("already in db");
+
+            mushroom.setId(databaseCounter);
+            database.put(databaseCounter++, mushroom);
+        }
+
+        public void update(Mushroom mushroom) {
+            validate(mushroom);
+            if (mushroom.getId() == null) throw new IllegalArgumentException("not persisted - cannot be updated");
+            if (database.replace(mushroom.getId(),mushroom) == null) throw new IllegalArgumentException("no object with such id in DB - cannot be updated");;
+        }
+
+        public void delete(Mushroom mushroom){
+            validate(mushroom);
+            if (mushroom.getId() == null) throw new IllegalArgumentException("no id assigned");
+            //TODO: deleting non persisted should be OK or NOK?
```

Comment:
same as in dao layer - should not throw an exception, but should not modify db (if i recall correctly)